modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java

```diff
@@ -356,8 +362,10 @@ private void initArgumentResolvers() {
           }
         }
       }
-    } else {
+    } else if (Object.class.equals(argumentType)) {
```

Comment:
MULE-17903 to verify why this was done, as indicated in the PR full explanation.